### ISSUE TITLE
Feture request: Expose ICellArray to Extensions 

### ISSUE THREAD

**Issue by Pelipoika** [alliedmodders/sourcemod#720]
- What is your end goal, or what are you trying to accomplish?

I want to expose https://github.com/ValveSoftware/source-sdk-2013/blob/master/mp/src/game/server/nav_pathfind.h#L751 (CollectSurroundingAreas) to SourceMod plugins like:

ArrayList array = CreateArray();
CollectSurroundingAreas(array, ....)

and the extension would then fill the array with the surrounding areas.

- Why is this necessary, or what benefit do you see with it?

It would be a massive pain in the ass to expose all the functions that are used in CollectSurroundingAreas to SourcePawn so you could manually construct the function in SourcePawn, instead of just simply being able to fill an array.

**Comment by arthurdead** [comment]
still need this.......

**Comment by rtldg** [comment]
Being able to extract the `ICellArray`/`CellArray` freely from a handle would be nice too. I'm currently doing hacky stuff to get `g_pCoreIdent` so `ReadHandle` works cleanly.

**Comment by BoneTM** [comment]
> Being able to extract the `ICellArray`/`CellArray` freely from a handle would be nice too. I'm currently doing hacky stuff to get `g_pCoreIdent` so `ReadHandle` works cleanly (`g_pCoreIdent` being exported in Windows dlls would be cool 👀).

did you get it, want to extend some native for the core handle type

**Comment by rtldg** [comment]
> did you get it, want to extend some native for the core handle type

yeah, this is a link to the file where I grab `g_pCoreIdent` and then ctrl+f how I used it https://github.com/rtldg/sm_closestpos/blob/f113dd3acdeae93caa76009896a48e42179c0fc1/extension.cpp#L132-L185

**Comment by BoneTM** [comment]
> > did you get it, want to extend some native for the core handle type
> 
> yeah, this is a link to the file where I grab `g_pCoreIdent` and then ctrl+f how I used it https://github.com/rtldg/sm_closestpos/blob/f113dd3acdeae93caa76009896a48e42179c0fc1/extension.cpp#L132-L185

ty so much, really need this, thanks again
